src/main/webapp/homepage_script.js

```diff
@@ -93,10 +108,85 @@ window.onclick = function(event) {
 }
 
 function adjustControlBar(userLogged){
-    console.log(userLogged);
 	if (userLogged === false) {
         document.getElementById("categories").style.width = "100%";
     } else {
         document.getElementById("categories").style.width = "90%";
     }
-}
\ No newline at end of file
+}
+
+/* Event listener to get user's neighborhood*/
+if (document.readyState === 'loading') {
+	document.addEventListener('DOMContentLoaded', getUserNeighborhood);
+} else {
+    getUserNeighborhood();
+}
```

Comment:
I think it might be better to add more comments for this block of code. It took me a while to understand what this block of code is doing. Also, will this part of code be called every time the page is loaded? If it is, I would prefer putting it into a function and call the function when the page loads.

Reply:
My bad, I'll update this with more comments to make it more readable! It's essentially adding an eventlistener to the document for it to call getUserNeighborhood when it loads. However, because the document might've loaded already by the time that script line is read, it first check if it's still loading. If it already loaded then the onload function will never be called, so instead it just goes ahead and calls it anyway.
It looks pretty odd (it did as well to me at first), but it seems to be the way to go according to this (see the 3rd example): https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event

Reply:
Ah I think you can just follow that example and add the two comments `// Loading hasn't finished yet` and `// `DOMContentLoaded` has already fired` to the if and else statement respectively. That will be easier for the reader to understand.